test/browser/async_pthread_join.c

```diff
@@ -0,0 +1,30 @@
+// Copyright 2023 The Emscripten Authors.  All rights reserved.
+// Emscripten is available under two separate licenses, the MIT license and the
+// University of Illinois/NCSA Open Source License.  Both these licenses can be
+// found in the LICENSE file.
+
+#include <emscripten.h>
+#include <pthread.h>
+#include <stdio.h>
+#include <assert.h>
+
+EM_ASYNC_JS(int, async_call, (), {
+  await new Promise((resolve) => setTimeout(resolve, 0));
+  return 42;
+});
+
+EMSCRIPTEN_KEEPALIVE void *run_thread(void *args) {
```

Comment:
Presumably we would like to avoid needing to add `EMSCRIPTEN_KEEPALIVE` here at some point?  Maybe add a TODO?